mongowire/wire_message.go

```diff
@@ -212,10 +216,10 @@ func (h *MessageHeader) parseMsgBody(body []byte) (Message, error) {
 			if err != nil {
 				return nil, errors.Wrap(err, "could not read identifier")
 			}
-			loc += len(section.Identifier)
+			loc += len(section.Identifier) + 1
 
-			for remaining := int(section.Size) - 4 - len(section.Identifier); remaining > 0; {
-				docSize := int(readInt32(body[loc+1:]))
+			for remaining := int(section.Size) - 1 - 4 - len(section.Identifier) - 1; remaining > 0; {
```

Comment:
it feels like the +1 on line 219 and then the -1 here could maybe be a bit clearer.

Reply:
its for the null terminator of c strings

Reply:
added a comment above